src/main/kotlin/events/autoroles.kt

```
@@ -0,0 +1,100 @@
+package events
+
+import core.AutoRoleDTO
+import core.getConfigOrNull
+import net.dv8tion.jda.api.EmbedBuilder
+import net.dv8tion.jda.api.JDA
+import net.dv8tion.jda.api.entities.Guild
+import net.dv8tion.jda.api.entities.TextChannel
+import net.dv8tion.jda.api.events.interaction.ButtonClickEvent
+import net.dv8tion.jda.api.hooks.ListenerAdapter
+import net.dv8tion.jda.api.interactions.components.ActionRow
+import net.dv8tion.jda.api.interactions.components.Button
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+val autoRoles: MutableMap<Long, MutableMap<String, AutoRole>> = mutableMapOf()
+val logger: Logger = LoggerFactory.getLogger("AutoRolesManager")
+
+fun createAutoRoleIfAbsent(guild: Guild, name: String, config: AutoRoleDTO): AutoRole? {
+    if (guild.idLong !in autoRoles.keys) {
+        autoRoles[guild.idLong] = mutableMapOf()
+    }
+
+    if (name !in autoRoles[guild.idLong]!!.keys) {
```

Never use !! outside of tests. Instead, keep autoRoles[guild.idLong] in a `val` beforehand and assert that it is not null. You can throw a proper exception if that happens to be the case. You should never assume the JVM will never have memory issues, thus, you can supposedly never be sure 100% something is not null.